### ISSUE TITLE
🐛 deconstructing does not satisfiy `This private class member is defined but never used.`

### ISSUE THREAD

**Issue by ViliamKopecky** [biomejs/biome#6974]
### Environment information

```block
CLI:
  Version:                      2.1.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.4
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.9.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

```ts
export class GoodExample {
	constructor(private something: string) {}

	example() {
		return this.something
	}
}


export class BadExample {
	constructor(private something: string) {
		//              ^^^^^^^^^ This private class member is defined but never used.
	}

	example() {
		const { something } = this
		return something
	}
}
```

Playground: https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABHAG8AbwBkAEUAeABhAG0AcABsAGUAIAB7AAoACQBjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABwAHIAaQB2AGEAdABlACAAcwBvAG0AZQB0AGgAaQBuAGcAOgAgAHMAdAByAGkAbgBnACkAIAB7AH0ACgAKAAkAZQB4AGEAbQBwAGwAZQAoACkAIAB7AAoACQAJAHIAZQB0AHUAcgBuACAAdABoAGkAcwAuAHMAbwBtAGUAdABoAGkAbgBnAAoACQB9AAoAfQAKAAoACgBlAHgAcABvAHIAdAAgAGMAbABhAHMAcwAgAEIAYQBkAEUAeABhAG0AcABsAGUAIAB7AAoACQBjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABwAHIAaQB2AGEAdABlACAAcwBvAG0AZQB0AGgAaQBuAGcAOgAgAHMAdAByAGkAbgBnACkAIAB7AH0ACgAKAAkAZQB4AGEAbQBwAGwAZQAoACkAIAB7AAoACQAJAGMAbwBuAHMAdAAgAHsAIABzAG8AbQBlAHQAaABpAG4AZwAgAH0AIAA9ACAAdABoAGkAcwAKAAkACQByAGUAdAB1AHIAbgAgAHMAbwBtAGUAdABoAGkAbgBnAAoACQB9AAoAfQAKAA%3D%3D

### Expected result

Should not report warning

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by vladimir-ivanov** [comment]
apologies, wont have time to work on this any time soon. Unassigned.
